src/fs/fs.c

```diff
@@ -92,94 +162,40 @@ void fs_ns_scan_cb(struct namespace *ns, size_t ns_size)
 {
 	str256_t *fs_name = NULL;
 	struct cfs_fs_node *fs_node = NULL;
+	int rc = 0;
 
 	/**
 	 * Before this fs can be added to the incore list fs_list, we must
-	 * prepare the incore structure to be usable by others, i.e.
-	 * the kvtree and namespace should be attached.
+	 * initialize the incore structure to be usable by others.
 	 */
 	ns_get_name(ns, &fs_name);
-	log_info("%d trying to load FS: " STR256_F,
-		 __LINE__, STR256_P(fs_name));
+	log_info("trying to load FS: " STR256_F,
+		 STR256_P(fs_name));
 
 	fs_node = calloc(1, sizeof(struct cfs_fs_node));
 	if (fs_node == NULL) {
-	    log_err("%d failed to load FS: " STR256_F
-		    " , could not allocate memory for cfs_fs_node!",
-		    __LINE__, STR256_P(fs_name));
-	    goto out_failed;
-	}
-
-	fs_node->cfs_fs.ns = calloc(1, ns_size);
-	if (fs_node->cfs_fs.ns == NULL) {
-	    log_err("%d failed to load FS: " STR256_F
-		    " , could not allocate memory for ns object!",
-		    __LINE__, STR256_P(fs_name));
-	    goto out_failed;
-	}
-
-	memcpy(fs_node->cfs_fs.ns, ns, ns_size);
-
-	fs_node->cfs_fs.kvtree = calloc(1, sizeof(struct kvtree));
-	if (!fs_node->cfs_fs.kvtree) {
-	    log_err("%d failed to load FS: " STR256_F
-		    " , could not allocate memory for kvtree object!",
-		    __LINE__, STR256_P(fs_name));
-	    goto out_failed;
-	}
-
-	if (kvtree_init(fs_node->cfs_fs.ns, fs_node->cfs_fs.kvtree) != 0) {
-	    log_err("%d failed to load FS: " STR256_F
-		    " , kvtree_init() failed!", __LINE__,
-		    STR256_P(fs_name));
-	    goto out_failed;
+		rc = -ENOMEM;
+	    goto out;
 	}
 
-	fs_node->cfs_fs.root_node = malloc(sizeof(struct kvnode));
-	if (!fs_node->cfs_fs.root_node) {
-		log_err("%d failed to load FS: " STR256_F
-			" , could not allocate memory for kvnode object!",
-			__LINE__, STR256_P(fs_name));
-		goto out_failed;
+	rc = fs_node_init(fs_node, ns, ns_size);
+	if (rc != 0) {
+		goto fs_init_fail;
 	}
 
-	*(fs_node->cfs_fs.root_node) = KVNODE_INIT_EMTPY;
-
-	if (kvnode_load(fs_node->cfs_fs.kvtree,
-			&(fs_node->cfs_fs.kvtree->root_node_id),
-			fs_node->cfs_fs.root_node) != 0) {
-            log_err("%d failed to load FS: " STR256_F
-                    " , kvnode_load() failed!", __LINE__,
-                    STR256_P(fs_name));
-            goto out_failed;
-        }
-
 	LIST_INSERT_HEAD(&fs_list, fs_node, link);
 	log_info("FS:" STR256_F " loaded from disk, ptr:%p",
 		 STR256_P(fs_name), &fs_node->cfs_fs);
 	return;
 
-out_failed:
-
-	log_info("FS:" STR256_F " failed to load from disk",
-		 STR256_P(fs_name));
-
-	if (fs_node) {
-		if (fs_node->cfs_fs.ns) {
-			free(fs_node->cfs_fs.ns);
-		}
-		if (fs_node->cfs_fs.root_node) {
-			/* kvnode_fini is a no-op if root_node is not loaded */
-			kvnode_fini(fs_node->cfs_fs.root_node);
-			free(fs_node->cfs_fs.root_node);
-		}
-		if (fs_node->cfs_fs.kvtree) {
-			kvtree_fini(fs_node->cfs_fs.kvtree);
-			free(fs_node->cfs_fs.kvtree);
-		}
-
-		free(fs_node);
+fs_init_fail:
+	free(fs_node);
+out:
+	if (rc != -ENOMEM) {
```

Comment:
I believe one of the point in ticket was if we get -ENOMEM then we should not be logging any information as it might fail. So still you want to log this?